consensus/src/epoch_manager.rs

```diff
@@ -309,16 +324,116 @@ impl EpochManager {
         Ok(())
     }
 
-    fn spawn_quorum_store(
+    fn spawn_direct_mempool_quorum_store(
         &mut self,
-        consensus_to_quorum_store_receiver: Receiver<ConsensusRequest>,
+        consensus_to_quorum_store_rx: Receiver<WrapperCommand>,
     ) {
         let quorum_store = DirectMempoolQuorumStore::new(
-            consensus_to_quorum_store_receiver,
-            self.quorum_store_to_mempool_sender.clone(),
+            self.quorum_store_to_mempool_tx.clone(),
             self.config.mempool_txn_pull_timeout_ms,
         );
+        // TODO: do we need to destroy the async thread with explicit shutdown?
+        tokio::spawn(quorum_store.start(consensus_to_quorum_store_rx));
+    }
+
+    ///this function spawns QuorumStore
+    fn spawn_quorum_store(
+        &mut self,
+        config: QuorumStoreConfig,
+        network_sender: NetworkSender,
+        verifier: ValidatorVerifier,
+        wrapper_command_rx: tokio::sync::mpsc::Receiver<QuorumStoreCommand>,
+    ) -> Arc<BatchReader> {
+        let backend = &self.config.safety_rules.backend;
+        let storage: Storage = backend.try_into().expect("Unable to initialize storage");
+        if let Err(error) = storage.available() {
+            panic!("Storage is not available: {:?}", error);
+        }
+        let private_key = storage
+            .export_private_key(CONSENSUS_KEY)
+            .expect("Unable to get private key");
+        let signer = ValidatorSigner::new(self.author, private_key);
```

Comment:
@zekun000 #1764 doesn't update `export_private_key(...)` to use BLS. Should it be updated? Or is there another recommended way to get the private key?

Reply:
Changing to `storage.get(CONSENSUS_KEY)...` I see on smoke tests:

```
2022-07-15T21:57:18.787506Z [consensus] ERROR crates/crash-handler/src/lib.rs:38 details = '''panicked at 'Unable to get private key: KeyNotSet("consensus")', consensus/src/epoch_manager.rs:410:14'''
backtrace = '''
```

Reply:
@zekun000 I see that smoke tests now use InMemoryStorage. This breaks the assumption here that reading from storage after `backend.try_into()` can read the private key (for InMemoryStorage this just creates a fresh HashMap).

I see the configs themselves have changed in the smoke tests.

Before:
```
safety_rules: SafetyRulesConfig { backend: OnDiskStorage(OnDiskStorageConfig
```
After:
```
safety_rules: SafetyRulesConfig { backend: InMemoryStorage
```

Is this an intended change? I couldn't figure out where exactly the config changed. I guess we should either make the InMemory store a singleton so the same HashMap is used, or just revert back to OnDisk.